### PR TITLE
[docs][s] Improving docs with example - fixes #14

### LINKED ISSUE #14: [docs][question]
If we're trying to set up a local server for test, what's the endpoint we should specify in the .lfsconfig file? "http://127.0.0.1:5000/"?

### PR DESCRIPTION
**Pull request by hannelita**
- Simple flow case to run it locally (create a project, track files with git LFS)
- Configure .lsfinfo, specifying endpoint
- List auth modes for anonymous (read_only/read_write)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -104,6 +108,47 @@ at runtime as an environment variable:
...
+### Running a local example
+
+First, create a new project on Github or any other platform.
+Here, we create `example-proj-datahub-io`. Add any data file
+to it. The goal is to track this possible large file with 
+git-lfs and use Giftless as the local server. In our example,
+we create a CSV named `research_data_factors.csv`.
+
+Next, start the Giftless server. Go to your local project 
+directory and run:
```
Comment (shevron) [inline README.md:121]: In terms of style I find that it is much easier to read an example as a list of steps (ordered or bullets) rather than a paragraph of text. The use of "first" and "next" is also confusing because these are used as filler words and not as actual step orders (next is not the next step after first in this description). In general I recommend rephrasing this section so that you state the goals and general things first, then break down the flow into steps. 

It is also worth breaking the shell code into sections with some explanations in the middle otherwise it becomes a huge blob of commands which is hard to follow.

File: README.md [hunk 2]
```diff
@@ -104,6 +108,47 @@ at runtime as an environment variable:
...
+
+```bash
+example-proj-datahub-io$ git init
+example-proj-datahub-io$ git remote add origin YOUR_REMOTE_REPO
+example-proj-datahub-io$ git lfs track '*.csv'
+example-proj-datahub-io$ git lfs track
```
Comment (shevron) [inline README.md:127]: Is it really necessary to have two `git lfs track` commands?

File: README.md [hunk 3]
```diff
@@ -104,6 +108,47 @@ at runtime as an environment variable:
...
+example-proj-datahub-io$ git init
+example-proj-datahub-io$ git remote add origin YOUR_REMOTE_REPO
+example-proj-datahub-io$ git lfs track '*.csv'
+example-proj-datahub-io$ git lfs track 
+example-proj-datahub-io$ git add .gitattributes #you should have a .gitattributes file at this point
+example-proj-datahub-io$ git add "*.csv"
+example-proj-datahub-io$ git commit -m "Tracking data files"
+example-proj-datahub-io$ git lfs ls-files # you should see your file listed
+example-proj-datahub-io$ git config -f .lfsconfig lfs.url http://127.0.0.1:5000/<user_or_org>/<repo>/
+# in our case, we used http://127.0.0.1:5000/datopian/example-proj-datahub-io/;
```
Comment (shevron) [inline README.md:133]: Please note that this repo is entirely unrelated to `datahub` or any other Datopian specific project and we should probably refrain from referencing anything specific in examples to avoid confusion.

Comment (shevron) [reply README.md:133]: It's better to use something like `myorg/myproject` for example

File: README.md [hunk 4]
```diff
@@ -104,6 +108,47 @@ at runtime as an environment variable:
...
+example-proj-datahub-io$ git add .gitattributes #you should have a .gitattributes file at this point
+example-proj-datahub-io$ git add "*.csv"
+example-proj-datahub-io$ git commit -m "Tracking data files"
+example-proj-datahub-io$ git lfs ls-files # you should see your file listed
+example-proj-datahub-io$ git config -f .lfsconfig lfs.url http://127.0.0.1:5000/<user_or_org>/<repo>/
+# in our case, we used http://127.0.0.1:5000/datopian/example-proj-datahub-io/; 
+# make sure to end your lfs.url with /
+example-proj-datahub-io$ git add .lfsconfig
+example-proj-datahub-io$ git commit -m "New git-lfs server endpoint" 
+# if you don't see any changes, run git rm --cached *.csv and then re-add your files, then commit it
```
Comment (shevron) [inline README.md:137]: Why would this happen? Is there some kind of magic?

File: README.md [hunk 5]
```diff
@@ -104,6 +108,47 @@ at runtime as an environment variable:
...
+example-proj-datahub-io$ git push origin master
+example-proj-datahub-io$ git lfs push origin master
```
Comment (shevron) [inline README.md:139]: I think `git lfs push` should happen automatically when pushing to origin... If this doesn't there is most likely a problem with the setup?

File: README.md [hunk 6]
```diff
@@ -104,6 +108,47 @@ at runtime as an environment variable:
...
+```
```
Comment (shevron) [inline README.md:140]: To reduce visual complexity it is better to avoid long prompts unless really necessary. 

```suggestion
```bash
$ git init
$ git remote add origin YOUR_REMOTE_REPO
$ git lfs track '*.csv'
$ git lfs track 
$ git add .gitattributes #you should have a .gitattributes file at this point
$ git add "*.csv"
$ git commit -m "Tracking data files"
$ git lfs ls-files # you should see your file listed
$ git config -f .lfsconfig lfs.url http://127.0.0.1:5000/<user_or_org>/<repo>/
# in our case, we used http://127.0.0.1:5000/datopian/example-proj-datahub-io/; 
# make sure to end your lfs.url with /
$ git add .lfsconfig
$ git commit -m "New git-lfs server endpoint" 
# if you don't see any changes, run git rm --cached *.csv and then re-add your files, then commit it
$ git push origin master
$ git lfs push origin master
```
```

File: README.md [hunk 7]
```diff
@@ -79,6 +79,10 @@ production environment:
 
     $ (.venv) ./flask-develop.sh
 
+The default generated endpoint is <http://127.0.0.1:5000/>. You should
+be able to access it, however you might receive an error message
+(the server will be empty).
```
[the remark this answers is not in the document]
Comment (hannelita) [reply README.md:84]: I think using _should_ is a bit more polite; but I will clarify this is not a valid route. I think it's good to clarify this, because many people may try to access the route they see in the logs.

File: README.md (hunk 2 above)
Comment (hannelita) [reply README.md:127]: The second one lists the tracked files. I can add a comment clarifying this

File: README.md (hunk 3 above)
Comment (hannelita) [reply README.md:133]: Sorry, I don't think `myorg/myproject` helps an entirely new developer. If the person only has their own Gitub account, it may not be clear what 'myorg' is. But surely I could use another string that is not 'datopian'. Note that I'm not using any reference to datahub though.

File: README.md (hunk 4 above)
Comment (hannelita) [reply README.md:137]: If you happen to run a slightly different sequence of commands you may need to re-add the files.

File: README.md [hunk 8]
```diff
@@ -104,6 +108,47 @@ at runtime as an environment variable:
...
+
+You might need to configure your `config.py` to accept non-authenticated reads writes:
+
+```python
+default_config = {
+    "AUTH_PROVIDERS": [
+        'giftless.auth.allow_anon:read_write'
+    ],
+    #...
+```
```
[the remark this answers is not in the document]
Comment (hannelita) [reply README.md:150]: I will change the docs then. It was not clear to me nor to @sglavoie ; I think we can make it more explicit in the config section.

File: README.md (hunk 1 above)
Comment (hannelita) [reply README.md:121]: It is a relatively short paragraph. No complex instructions in it and the content may vary. I think users may have no problem following it. I don't think 'first' and 'next' are filler words, they just indicate a logical sequence to be followed

Comment (rufuspollock) [reply README.md:121]: I would 👍 @shevron comment on bullet points.

File: README.md (hunk 2 above)
Comment (rufuspollock) [reply README.md:127]: @hannelita i would not track *.csv but rather track a specific csv files atm ...

File: README.md (hunk 4 above)
Comment (rufuspollock) [reply README.md:137]: I think we should keep this as simple as possible and cover possible errors in FAQs or similar.

File: README.md (hunk 6 above)
Comment (rufuspollock) [reply README.md:140]: Big 👍

**Comment by hannelita** [comment]
@shevron @rufuspollock I think I addressed all your comments. Thanks for the feedback!
